### PR TITLE
Switch to specific campaigns based on then url

### PR DESCRIPTION
**Pull request by gabriellui1**
fixes ENG-1068
The url for the campaign tabs is `teachers/curriculum/{campaign-slug}`
image

## Summary by CodeRabbit

- **New Features**
	- Introduced dynamic routing for teachers to access campaign-specific curriculum.
	- Enhanced curriculum guide component to accept and utilize a campaign URL for improved navigation and context-aware rendering.
	- Added options for flexible data fetching in the teacher dashboard, allowing for context-sensitive behavior based on campaign URLs.

- **Bug Fixes**
	- Improved logic to prevent unnecessary executions when selecting campaigns based on the current course context.

### CODE REVIEW

**Review by smallst — APPROVED** [review]
didn't test bug LGTM

File: ozaria/site/components/teacher-dashboard/BaseCurriculumGuide/components/ChapterNav.vue [hunk 1]
```diff
@@ -32,6 +46,21 @@ export default {
     }
   },
 
+  watch: {
+    getCurrentCourse () {
```
Comment (mrfinch) [inline ozaria/site/components/teacher-dashboard/BaseCurriculumGuide/components/ChapterNav.vue:50]: any reason for doing `watch` over just setting the selected campaign in `setDefaultCampaign`?

Comment (gabriellui1) [reply ozaria/site/components/teacher-dashboard/BaseCurriculumGuide/components/ChapterNav.vue:50]: The junior campaign is set as default somewhere else after `setDefaultCampaign` is ran. Using watch is necessary to switch the campaign after the campaign defaults to junior.

Comment (mrfinch) [reply ozaria/site/components/teacher-dashboard/BaseCurriculumGuide/components/ChapterNav.vue:50]: any idea where it is set like that?

**Review by mrfinch — COMMENTED** [review]
fetchDataCurriculumGuide in teacherDashboard store could be modified to pass campaignId so that default selected is appropriate

**Review by mrfinch — APPROVED** [review]

**Pull request merged**